tests/test_launch.py

```diff
@@ -43,63 +52,196 @@ def tearDown(self) -> None:
+                        "prev_tasks": [len(sample_names)],
+                        "parameters": {},
+                        "samples": [sample_name_],
+                    } for sample_name_ in sample_names],
+
+                ],
+            }
+
+        exp_ids = []
+        num_of_tasks = 0
+        for i in range(8):
+            experiment = compose_exp(f"Experiment with {i+1} samples", num_samples=i+1)
+            num_of_tasks += len(experiment["tasks"])
+            resp = requests.post(
+                "http://127.0.0.1:8896/api/experiment/submit", json=experiment
+            )
+            resp_json = resp.json()
+            exp_id = ObjectId(resp_json["data"]["exp_id"])
+            self.assertTrue("success", resp_json["status"])
+            exp_ids.append(exp_id)
+        time.sleep(30)
+        self.assertEqual(num_of_tasks, self.task_view._task_collection.count_documents({}))
+
+        self.assertTrue(
+            all(
+                task["status"] == "COMPLETED"
+                for task in self.task_view._task_collection.find()
+            )
+        )
+        self.assertTrue(
+            all(
+                task["result"] == task["_id"]
+                for task in self.task_view._task_collection.find()
+            )
+        )
+
+        for exp_id in exp_ids:
+            self.assertEqual(
+                "COMPLETED", self.experiment_view.get_experiment(exp_id)["status"]
+            )
+
+    def test_user_input(self):
+        def compose_exp(exp_name, error_task):
+            return {
+                "name": exp_name,
+                "tags": [],
+                "metadata": {},
+                "samples": [{"name": "test_sample", "tags": [], "metadata": {}}],
+                "tasks": [
+                    {
+                        "type": "Starting",
+                        "prev_tasks": [],
+                        "parameters": {
+                            "dest": "furnace_table",
+                        },
+                        "samples": ["test_sample"],
                     },
-                    "samples": ["test_sample"],
-                },
-                {
-                    "type": "Ending",
-                    "prev_tasks": [1],
-                    "parameters": {},
-                    "samples": ["test_sample"],
-                },
-            ],
-        }
+                    {
+                        "type": error_task,
+                        "prev_tasks": [0],
+                        "parameters": {},
+                        "samples": ["test_sample"],
+                    },
+                ],
+            }
         exp_ids = []
-        for _ in range(3):
+        for error_name in ["ErrorHandlingUnrecoverable", "ErrorHandlingRecoverable"]:
+            experiment = compose_exp(f"Experiment with {error_name}", error_task=error_name)
             resp = requests.post(
                 "http://127.0.0.1:8896/api/experiment/submit", json=experiment
```

Comment:
"http://127.0.0.1:8896/api/experiment/submit" -> you use it often ,is not it better to be global?

Reply:
added